hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/commit/BaseCommitActionExecutor.java

```diff
@@ -200,4 +215,65 @@ protected boolean isWorkloadProfileNeeded() {
 
   protected abstract Iterator<List<WriteStatus>> handleUpdate(String partitionPath, String fileId,
       Iterator<HoodieRecord<T>> recordItr) throws IOException;
+
+  protected HoodieWriteMetadata<HoodieData<WriteStatus>> executeClustering(HoodieClusteringPlan clusteringPlan) {
+    HoodieInstant instant = HoodieTimeline.getReplaceCommitRequestedInstant(instantTime);
```

Comment:
The `BaseCommitActionExecutor` responsibilities are a bit confusing, it handles regular writing process such as `insert`, `upsert` and with this path `clustering`, then what about the `compaction`?

Should we make a new base class for table services then ?

Reply:
I think the goal is to revamp the commit executors and write pipeline altogether later on, so the refactoring here is limited to code reuse.  @xushiyan is that the case?

Reply:
@danny0405 agreed that it looks like some mixed responsibilities there. i'll make clearer separation in https://issues.apache.org/jira/browse/HUDI-2439